### PR TITLE
Adapt SNMP flow component/integration to new SNMP exporter config changes

### LINKED ISSUE #3421: Get changes of `snmp_exporter` merged upstream and remove `replace` in go.mod

### LINKED ISSUE #3843: Allow for prometheus.exporter.snmp to get the config file from other components
### Request

Currently, the `prometheus.exporter.snmp' component needs a `config_file` argument, which the [docs](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.exporter.snmp) say is a path to a file.

This ends up being inconvenient, especially on a Kubernetes deployment of the agent, which would require me to mount that config file on the agent pod somewhere. It would be nice if that config could come from a different component, like `local.file` or `remote.http`.

### Use case

This would provide flexibility in how to configure a Grafana Agent to scrape SNMP metrics.

### PR DESCRIPTION
**Pull request by marctc**
#### PR Description

This PR introduces several changes that affect SNMP flow component and integration:

- Add the last version of SNMP exporter which includes a [new way to configure it](https://github.com/prometheus/snmp_exporter/pull/859) . This makes breaking changes in `walk_params` struct of the exporter removing `auth` and `version`, but at the same time allows flexibility to uses to define `auth` at config level.
- Add the ability to define SNMP config as embedded yaml in SNMP flow component and integration. This is specially useful to use in Flow with `local.file` component.
- Refactor some `Handler` method to avoid code duplications.
- Remove dependency of custom fork to avoid registration of metrics to the global registry.

#### Which issue(s) this PR fixes

Fixes #3421 
Fixes #3843 

#### PR Checklist

- [x] CHANGELOG updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,13 @@ internal API changes are not present.
 Main (unreleased)
 -----------------
 
+> **BREAKING CHANGES**: This release has breaking changes. Please read entries
+> carefully and consult the [upgrade guide][] for specific instructions.
+
+### Breaking changes
+
+- The field `version` and `auth` struct block from `walk_params` in `prometheus.exporter.snmp` and SNMP integration have been removed. The auth block now can be configured at top level, together with `modules` (@marctc)
```
Comment (mattdurham) [inline CHANGELOG.md:18]: Can we automatically transform it so its not a breaking change?

File: docs/sources/flow/reference/components/prometheus.exporter.snmp.md [hunk 2]
```diff
@@ -36,10 +36,18 @@ Omitted fields take their default values.
 
 Name | Type | Description | Default | Required
 ---- | ---- | ----------- | ------- | --------
-`config_file` | `string`       | SNMP configuration file defining custom modules. | | yes
+`config_file` | `string`       | SNMP configuration file defining custom modules. | | no
+`config` | `string`       | SNMP configuration as inline string.  | |no
```
Comment (mattdurham) [inline docs/sources/flow/reference/components/prometheus.exporter.snmp.md:40]: Big thumbs up to `config` being an inline string

**Review by mattdurham — APPROVED** [review]
LGTM, this may get you both the 2023 winner for lines removed and added =)

**Review by clayton-cornell — APPROVED** [review]
Docs look OK to me

**Pull request merged**